scripts/Imaging/imgpy/tools/tomopy_tool.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import (absolute_import, division, print_function)
+from tools.abstract_tool import AbstractTool
+import helper as h
+
+
+class TomoPyTool(AbstractTool):
+    @staticmethod
+    def tool_supported_methods():
+        return [
+            'art', 'bart', 'fbp', 'gridrec', 'mlem', 'osem', 'ospml_hybrid',
+            'ospml_quad', 'pml_hybrid', 'pml_quad', 'sirt'
+        ]
+
+    @staticmethod
+    def check_algorithm_compatibility(algorithm):
+        if algorithm not in TomoPyTool.tool_supported_methods():
+            raise ValueError(
+                "The selected algorithm {0} is not supported by TomoPy.".
+                format(algorithm))
+
+    def __init__(self):
+        AbstractTool.__init__(self)
+        self._tomopy = self.import_self()
+        import tomopy.prep
+        import tomopy.recon
+        import tomopy.misc
+        import tomopy.io
+        import tomopy.sim
+
+        # pretend we have the functions
+        self.find_center = self._tomopy.find_center
+        self.find_center_vo = self._tomopy.find_center_vo
+        self.circ_mask = self._tomopy.circ_mask
+
+        # make all tomopy methods available
+        self.misc = tomopy.misc
+        self.prep = tomopy.prep
+        self.recon = tomopy.recon
+        self.sim = tomopy.sim
+        self.io = tomopy.io
+
+    def import_self(self):
+        try:
+            import tomopy
+            import tomopy.prep
+            import tomopy.recon
+            import tomopy.misc
+            import tomopy.io
+
+        except ImportError as exc:
+            raise ImportError(
+                "Could not import the tomopy package and its subpackages. Details: {0}".
+                format(exc))
+
+        return tomopy
+
+    def run_reconstruct(self, sample, config, proj_angles=None, **kwargs):
+        """
+        Run a reconstruction with TomoPy, using the CPU algorithms they provide.
+
+        Information for each reconstruction method is available at
+            http://tomopy.readthedocs.io/en/latest/api/tomopy.recon.algorithm.html
+
+        :param sample: The sample image data as a 3D numpy.ndarray
+        :param config: A ReconstructionConfig with all the necessary parameters to run a reconstruction.
+        :param proj_angles: The projection angle for each slice
+        :param kwargs: Any keyword arguments will be forwarded to the TomoPy reconstruction function
+        :return: The reconstructed volume
+        """
+        import numpy as np
+        h.check_config_integrity(config)
+        h.check_data_stack(sample)
+
+        if proj_angles is None:
```

Comment:
I'm still not sure if this will be a python list or a numpy.array, I've added a comment for future me to decide